### PR TITLE
dts: arm: nxp: Add generated LPC55S69 DTS files

### PR DESCRIPTION
**Pull request by danieldegrasse**
Add generated LPC55S69 DTS files. This is an attempt to generate
the devicetree for the LPC55S69 automatically, in order to minimize
errors associated with manual DTS editing.

I'm opening this PR primarily to gather feedback on the approach here, in particular including compatibles for all SOC IP blocks.

### CODE REVIEW

**Review by MaureenHelm — COMMENTED** [review]
Are you generating the devicetrees from an internal database or from something public in MCUXpresso? How will contributors outside of NXP add SoCs?

**Comment by dleach02** [comment]
> Are you generating the devicetrees from an internal database or from something public in MCUXpresso? How will contributors outside of NXP add SoCs?

This is being generated from an internal database. The objective is to get away from the hand generated SOC dts which has historically been error prone. Presumably, the internal database, which is used to auto-generate a number of assets is considered the "source of truth" about the part.

This doesn't prevent outside contributors of NXP SoCs and if that SoCs is covered by our internal database we can consider facilitating the external contribution with support from the auto-generation.

File: dts/arm/nxp/nxp_lpc55S6x_ns.dtsi [hunk 1]
```diff
@@ -1,19 +1,558 @@
...
 		};
 
 		peripheral: peripheral@40000000 {
-			ranges = <0x0 0x40000000 0x10000000>;
+			#address-cells = <0x1>;
+			#size-cells = <0x1>;
+
+			ranges = <0x40000000 0x40000000 DT_SIZE_K(262144)>;
+
+			adc0: lpadc@400a0000 {
```
[the remark this answers is not in the document]
Comment (dleach02) [reply dts/arm/nxp/nxp_lpc55S6x_ns.dtsi:92]: Do you think that address would be preferable over name?

File: dts/arm/nxp/nxp_lpc55S6x_ns.dtsi [hunk 2]
```diff
@@ -1,19 +1,558 @@
...
+			};
+
+			sysctl: sysctl1@40023000 {
+				compatible = "nxp,sysctl1";
+				status = "disabled";
+				reg = <0x40023000 0x120>;
+			};
+
+			usb0: usb@40084000 {
+				compatible = "nxp,usb";
```
[the remark this answers is not in the document]
Comment (danieldegrasse) [reply dts/arm/nxp/nxp_lpc55S6x_ns.dtsi]: I've updated the compatible property to use the SDK driver name, versus the module name. This should create more unique compatibles.

File: dts/arm/nxp/nxp_lpc55S6x.dtsi [hunk 3]
```diff
@@ -1,36 +1,27 @@
 /*
- * Copyright (c) 2019, NXP
+ * Copyright (c) 2019,2023 NXP
  *
  * SPDX-License-Identifier: Apache-2.0
```
[the remark this answers is not in the document]
Comment (danieldegrasse) [reply dts/arm/nxp/nxp_lpc55S6x.dtsi:4]: Added a note about autogeneration.
> Also, secure dtsi vs the ns one

Do you want a comment in the devicetree specifying if the output is for a secure versus nonsecure core here?

File: dts/arm/nxp/nxp_lpc55S6x_ns.dtsi (hunk 1 above)
[the remark this answers is not in the document]
Comment (henrikbrixandersen) [reply dts/arm/nxp/nxp_lpc55S6x_ns.dtsi]: I'd prefer sorting by address as well.

File: dts/arm/nxp/nxp_lpc55S6x_ns.dtsi [hunk 4]
```diff
@@ -1,19 +1,562 @@
 /*
- * Copyright (c) 2019, NXP
+ * Copyright (c) 2019,2023 NXP
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/*
+ * This file is automatically generated based on the core
+ * definition for LPC55S69.
```
Comment (dleach02) [inline dts/arm/nxp/nxp_lpc55S6x_ns.dtsi:9]: Let us consider adding the tool version used to generate

File: dts/arm/nxp/nxp_lpc55S6x_ns.dtsi [hunk 5]
```diff
@@ -1,19 +1,568 @@
...
+				interrupts = <38 0>;
+			};
+
+			flash: flash@40034000 {
+				compatible = "nxp,iap-fmc55";
+				status = "disabled";
+				reg = <0x40034000 0x101c>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+				status = "disabled";
```
Comment (MaureenHelm) [inline dts/arm/nxp/nxp_lpc55S6x_ns.dtsi:261]: duplicate `status = "disabled"`

**Review by MaureenHelm — APPROVED** [review]
Looks good

**Pull request closed without merging**